src/windowstate.rs

```diff
@@ -124,15 +129,51 @@ impl WindowState {
 
     pub fn draw_frameless(&self) {
         self.connection.grab_server().unwrap();
-        self.connection.unmap_window(self.frame).unwrap();
-        self.connection.unmap_window(self.window).unwrap();
 
-        let window_aux = ConfigureWindowAux::new()
+        let frame_aux = ConfigureWindowAux::new()
             .x(self.x)
             .y(self.y)
             .width(self.width)
             .height(self.height);
 
+        let window_aux = ConfigureWindowAux::new()
+            .x(0)
+            .y(0)
+            .width(self.width)
+            .height(self.height);
+
+        self.connection
+            .configure_window(self.frame, &frame_aux)
+            .unwrap();
+        self.connection
+            .configure_window(self.window, &window_aux)
+            .unwrap();
+
+        self.connection.map_window(self.frame).unwrap();
+        self.connection.map_window(self.window).unwrap();
+
+        self.connection.ungrab_server().unwrap();
+        self.connection.flush().unwrap();
+    }
+
+    pub fn draw_without_border(&self){
```

Comment:
why do we have three functions?

I would think we only need two. One normal ones with gaps and frames, for fullscreen and status bar.
Another one with gaps and frames for normal windows.